node/config/types.go

```diff
@@ -590,3 +595,39 @@ type Wallet struct {
 type FeeConfig struct {
 	DefaultMaxFee types.FIL
 }
+
+// ClusterRaftConfig allows to configure the Raft Consensus component for the node cluster.
+type ClusterRaftConfig struct {
+	// config to enabled node cluster with raft consensus
+	ClusterModeEnabled bool
+	// will shutdown libp2p host on shutdown. Useful for testing
+	HostShutdown bool
+	// A folder to store Raft's data.
+	DataFolder string
+	// InitPeerset provides the list of initial cluster peers for new Raft
+	// peers (with no prior state). It is ignored when Raft was already
+	// initialized or when starting in staging mode.
+	InitPeerset []peer.ID
+	// LeaderTimeout specifies how long to wait for a leader before
+	// failing an operation.
+	WaitForLeaderTimeout time.Duration
+	// NetworkTimeout specifies how long before a Raft network
+	// operation is timed out
+	NetworkTimeout time.Duration
+	// CommitRetries specifies how many times we retry a failed commit until
+	// we give up.
+	CommitRetries int
+	// How long to wait between retries
+	CommitRetryDelay time.Duration
+	// BackupsRotate specifies the maximum number of Raft's DataFolder
+	// copies that we keep as backups (renaming) after cleanup.
+	BackupsRotate int
+	// Namespace to use when writing keys to the datastore
+	DatastoreNamespace string
+
+	// A Hashicorp Raft's configuration object.
+	RaftConfig *hraft.Config
```

Comment:
We likely want to "unpack" this config here

Reply:
It maybe be better to start with the default config always and then unpack this after. I feel usage might be a little bit more overwhelming if people have to start understanding the intricacies of raft immediately?

Reply:
That sounds reasonable